overlord/snapstate/handlers.go

```diff
@@ -1174,6 +1174,26 @@ func (m *SnapManager) doCopySnapData(t *state.Task, _ *tomb.Tomb) error {
 		if err != nil {
 			return fmt.Errorf("cannot set migration status to done (migration won't be undone if change fails): %w", err)
 		}
+	} else if !opts.UseHidden && opts.MigratedToHidden {
+		// migration was done but user turned the feature off, so undo migration
+		if err := m.backend.UndoHideSnapData(snapsup.InstanceName()); err != nil {
+			if err := m.backend.HideSnapData(snapsup.InstanceName()); err != nil {
+				st.Lock()
+				t.Logf("cannot revert snap dir unhiding (must manually restore %s's dirs from %s to %s): %v",
+					snapsup.InstanceName(), dirs.HiddenSnapDataHomeDir, dirs.UserHomeSnapDir, err)
+				st.Unlock()
+			}
+
+			return err
+		}
+
+		snapsup.MigratedExposed = true
+		st.Lock()
+		err := SetTaskSnapSetup(t, snapsup)
+		st.Unlock()
+		if err != nil {
+			return fmt.Errorf("cannot set migration status to undone (migration won't be redone if change fails): %w", err)
+		}
```

Comment:
What is the scenario where the change *doesn't fail*? Is it when lanes are used?

Reply:
I think that was meant as a "if we continued and failed (...)" but you're right, it's very confusing. It might be better to keep just the first part of the error message